File: src/com/example/main/filesystem/java/File.java
package com.example.main.filesystem;

public class File extends AbstractFileSystemNode {
    public File(String name, String path) {
        super(name, path);
    }

    public String getExtension() {
        String[] nameSplit = name.split("\\.");
        return (nameSplit.length > 1) ? nameSplit[1] : null;
    }
}

File: src/com/example/main/homework3/filesystem/File.kt
package com.example.main.homework3.filesystem

class File(_name: String, _path: String) : AbstractFileSystemNode(_name, _path) {

    fun getExtension(): String {
        val nameSplit = _name.split(".")
        return if (nameSplit.size > 1) nameSplit[1] else ""
    }
}